crates/katana/src/main.rs

```diff
@@ -102,3 +108,20 @@ ACCOUNTS SEED
 
     println!("\n{address}\n\n");
 }
+
+pub async fn shutdown_handler(sequencer: Arc<impl Sequencer>, config: KatanaArgs) {
+    println!(); // Extra line break for ^C keypress message
+    println!("Katana shutting down.");
+
+    if let Some(path) = config.dump_state {
+        let state = (*sequencer).backend().dump_state().await.unwrap();
```

Comment:
lets make the whole function infallible and not panic if something bad happen. just log out the error